### ISSUE TITLE
querydsl-jpa HibernateQuery in hibernate 5.4.0.Final not working

### ISSUE THREAD

**Issue by andreilocks** [querydsl/querydsl#2411]
When I make a filter with HibernateQuery the applciation throw the exception IllegalArgumentException.

java.lang.IllegalArgumentException: Could not locate named parameter [3], expecting one of []
	at org.hibernate.query.internal.ParameterMetadataImpl.getNamedParameterDescriptor(ParameterMetadataImpl.java:218) ~[hibernate-core-5.4.0.Final.jar:5.4.0.Final]
	at org.hibernate.query.internal.ParameterMetadataImpl.getQueryParameter(ParameterMetadataImpl.java:187) ~[hibernate-core-5.4.0.Final.jar:5.4.0.Final]
	at org.hibernate.query.internal.QueryParameterBindingsImpl.getBinding(QueryParameterBindingsImpl.java:188) ~[hibernate-core-5.4.0.Final.jar:5.4.0.Final]
	at org.hibernate.query.internal.AbstractProducedQuery.setParameter(AbstractProducedQuery.java:497) ~[hibernate-core-5.4.0.Final.jar:5.4.0.Final]
	at org.hibernate.query.internal.AbstractProducedQuery.setParameter(AbstractProducedQuery.java:110) ~[hibernate-core-5.4.0.Final.jar:5.4.0.Final]
	at com.querydsl.jpa.hibernate.HibernateUtil.setValue(HibernateUtil.java:91) ~[querydsl-jpa-4.1.4.jar:na]
	at com.querydsl.jpa.hibernate.HibernateUtil.setConstants(HibernateUtil.java:79) ~[querydsl-jpa-4.1.4.jar:na]

-----------------------------------------------------------------------------------------------------------------------------------

In HibernateUtil, on method *setValue(Query query, String key, Object val)*, on the else clouse have a query.setParameter(key, val). Same method in JPA have parse de key to Integer.

--------------------------------------------------------------------------------------------------------------------------------------
```
hQuery.select(Projections.fields(TotalPorAnoProjection.class,
                        debito.ano,
                        TOTAL_DEVIDO.sum().as("vlTotal")))
                .where(DebitoRepresentation.getVencidoCriteria(vencido),
                        ano != null ? debito.ano.eq(ano) : debito.ano.isNotNull(),
                        debito.situacao.eq(SituacaoParcela.ABERTO),
                        DebitoRepresentation.getFormaPagamentoCriteria())
                .groupBy(debito.ano)
                .orderBy(debito.ano.desc())
                .fetch();
```

**Comment by jfcabral** [comment]
Seems to duplicate #2326

**Comment by idosal** [comment]
Fixed in [#2354 ](https://github.com/querydsl/querydsl/pull/2354). It will be featured in the next official release, which will remove support for Hibernate 4. For now, you can try it out in 4.2.3-SNAPSHOT.
